sycl/source/builtins_generator.py

```diff
@@ -0,0 +1,1285 @@
+  "vgeodouble" : vgeodouble,
+  "vgeohalf" : vgeohalf,
+  "mgeofloat" : mgeofloat,
+  "mgeodouble" : mgeodouble,
+  "mgeohalf" : mgeohalf,
+  "gengeofloat" : gengeofloat,
+  "gengeodouble" : gengeodouble,
+  "gengeohalf" : gengeohalf,
+  "vint8n" : vint8n,
+  "vint16n" : vint16n,
+  "vint32n" : vint32n,
+  "vint64n" : vint64n,
+  "vuint8n" : vuint8n,
+  "vuint16n" : vuint16n,
+  "vuint32n" : vuint32n,
+  "vuint64n" : vuint64n,
+  "mint8n" : mint8n,
+  "mint16n" : mint16n,
+  "mint32n" : mint32n,
+  "muint8n" : muint8n,
+  "muint16n" : muint16n,
+  "muint32n" : muint32n,
+  "mintn" : mintn,
+  "mushortn" : mushortn,
+  "muintn" : muintn,
+  "mulongn" : mulongn,
+  "mbooln" : mbooln,
+  "geninteger" : geninteger,
+  "sigeninteger" : sigeninteger,
+  "vigeninteger" : vigeninteger,
+  "migeninteger" : migeninteger,
+  "igeninteger" : igeninteger,
+  "vugeninteger" : vugeninteger,
+  "ugeninteger" : ugeninteger,
+  "igenint32" : igenint32,
+  "ugenint32" : ugenint32,
+  "sgentype" : sgentype,
+  "vgentype" : vgentype,
+  "mgentype" : mgentype,
+  "intptr" : intptr,
+  "floatptr" : floatptr,
+  "doubleptr" : doubleptr,
+  "halfptr" : halfptr,
+  "vgenfloatptr" : vgenfloatptr,
+  "mgenfloatptr" : mgenfloatptr,
+  "mintnptr" : mintnptr,
+  "vint32nptr" : vint32ptr,
+  "elementtype0" : elementtype0,
+  "unsignedtype0" : unsignedtype0,
+  "samesizesignedint0" : samesizesignedint0,
+  "samesizeunsignedint0" : samesizeunsignedint0,
+  "samesizefloat0" : samesizefloat0,
+  "intelements0" : intelements0,
+  "upsampledint0" : upsampledint0,
+  "boolelements0" : boolelements0,
+  "char" : ["char"],
+  "signed char" : ["signed char"],
+  "short" : ["short"],
+  "int" : ["int"],
+  "long" : ["long"],
+  "long long" : ["long long"],
+  "unsigned char" : ["unsigned char"],
+  "unsigned short" : ["unsigned short"],
+  "unsigned int" : ["unsigned int"],
+  "unsigned long" : ["unsigned long"],
+  "unsigned long long" : ["unsigned long long"],
+  "float" : ["float"],
+  "double" : ["double"],
+  "half" : ["half"],
+  "int8_t" : ["int8_t"],
+  "int16_t" : ["int16_t"],
+  "int32_t" : ["int32_t"],
+  "int64_t" : ["int64_t"],
+  "uint8_t" : ["uint8_t"],
+  "uint16_t" : ["uint16_t"],
+  "uint32_t" : ["uint32_t"],
+  "uint64_t" : ["uint64_t"],
+  "bool" : ["bool"]
+}
+
+### BUILTINS
+
+# Finds the first templated argument in a type list.
+def find_first_template_arg(arg_types):
+  for arg_type in arg_types:
+    if isinstance(arg_type, InstantiatedTemplatedArg):
+      return arg_type
+  return None
+
+# Returns true if the argument type is a templated marray.
+def is_marray_arg(arg_type):
+  return isinstance(arg_type, InstantiatedTemplatedArg) and isinstance(arg_type.templated_type, Marray)
+
+# Returns true if the argument type is a templated vector.
+def is_vec_arg(arg_type):
+  return isinstance(arg_type, InstantiatedTemplatedArg) and isinstance(arg_type.templated_type, Vec)
+
+# Converts an argument name to a valid argument to be passed to another
```

Comment:
Should we use the official [docstring format](https://peps.python.org/pep-0257/) instead?